returnn/tf/util/data.py

```diff
@@ -824,48 +824,35 @@ def copy_with_time_dim_axis(self, time_dim_axis):
     assert self.time_dim_axis is not None
     return self.copy_move_axis(self.time_dim_axis, time_dim_axis)
 
-  def copy_move_axis(self, old_axis, new_axis):
+  def copy_transpose(self, perm):
     """
-    :param int old_axis: counted with batch-dim
-    :param int new_axis: counted with batch-dim
-    :return: copy of myself with moved axis (see :func:`move_axis`)
+    :param list[int] perm: permutation of the axes, counted with batch-dim.
+      Maps the new axes to the old axes
+    :return: copy of myself with permuted axes
     :rtype: Data
     """
-    from .basic import move_axis
-    if old_axis < 0:
-      old_axis += self.batch_ndim
-      assert old_axis >= 0
-    assert 0 <= old_axis < self.batch_ndim
-    if new_axis < 0:
-      new_axis += self.batch_ndim
-      assert new_axis >= 0
-    assert 0 <= new_axis < self.batch_ndim
-    if old_axis == new_axis:
+    assert len(perm) == self.batch_ndim
+    assert set(perm) == set(range(self.batch_ndim))
+    if all(perm[axis] == axis for axis in range(self.batch_ndim)):
       return self.copy()
+    inv_perm = [[new_axis for new_axis in range(self.batch_ndim) if perm[new_axis] == old_axis][0]
+      for old_axis in range(self.batch_ndim)]
```

Comment:
```suggestion
    inv_perm = [
      [new_axis for new_axis in range(self.batch_ndim) if perm[new_axis] == old_axis][0]
      for old_axis in range(self.batch_ndim)]
```
(That might fix also the indent warning.)

However, I think this code is too difficult to read. I would use sth like:
```suggestion
    perm_ = dict(enumerate(perm))
    inv_perm_ = {j: i for (i, j) in perm_.items()}
    inv_perm = [j for (i, j) in sorted(inv_perm_.items())]
```

Reply:
Thanks! `dict(enumerate(perm))` also gave me a PyCharm warning, [apparently a false positive](https://stackoverflow.com/questions/61792979/pycharm-warning-unexpected-type-for-filter-on-dictionary). I ended up using `enumerate(perm)` directly instead of `perm_.items()`

Reply:
Yea I already reported that problem upstream. (You also should always report any PyCharm problems you encounter.) But it should be ignored in our CI test case.